test/parallel/test-http2-compat-serverresponse-destroy.js

```diff
@@ -0,0 +1,91 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+// Check that destroying the Http2ServerResponse stream produces
+// the expected result, including the ability to throw an error
+// which is emitted on server.streamError
+
+const errors = [
+  'test-error',
+  Error('test')
+];
+let nextError;
+
+const server = http2.createServer(common.mustCall((req, res) => {
+  req.once('error', common.mustNotCall());
+  req.once('error', common.mustNotCall());
```

Comment:
this are both `req`. I think `res` should receive the error, considering the semantics of `.destroy(err)` throughout the codebase. Check with H1, but I think we have a bug in the compat layer because of this.